index.js

```diff
@@ -58,16 +75,20 @@ class ExecutorQueue extends Executor {
      * @return {Promise}
      */
     _stop(config) {
-        return this.breaker.runCommand('connect')
-            .then(() => this.breaker.runCommand('del', this.buildQueue, 'start', [config]))
+        return this.queueBreaker.runCommand('connect')
+            .then(() => this.queueBreaker.runCommand('del', this.buildQueue, 'start', [{
+                buildId: config.buildId,
+                annotations: config.annotations || ''
+            }]))
             .then((numDeleted) => {
                 if (numDeleted !== 0) {
                     // Build hadn't been started, "start" event was removed from queue
-                    return null;
+                    return this.redisBreaker.runCommand('hdel', this.buildConfigTable,
```

Comment:
worker also needs to ensure that it does `hdel`  after processing is done for the buildId